CMakeLists.txt

```diff
@@ -0,0 +1,85 @@
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the Swift open source project
+##
+## Copyright (c) 2024 Apple Inc. and the Swift project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.md for the list of Swift project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+cmake_minimum_required(VERSION 3.22)
```

Comment:
I have my suspicion that 3.22 does not actually work. Have you tested this build with CMake 3.22?
There were some fairly important bug fixes that went into CMake 3.25/3.26 for supporting projects that had targets building Swift + non-Swift and linking them into eachother.

Reply:
We've been testing locally with 3.24 which is what building the toolchain requires (swiftpm now uses features only available in 3.24). Should we bump this to 3.24 instead (since we know it works with that version and that's what the toolchain builds use)?

Reply:
I would certainly feel more comfortable saying we support something that we've tested, yes :)

It may also be interesting trying to build this with newer versions as well. e.g
```cmake
cmake_minimum_required(VERSION 3.24...)
```
CMake 3.29 had a fairly big change to how it models Swift builds (that helps a tad with build-perf), so it would be nice to test how it handles too.

Reply:
Sounds good we can update it to 3.24. As for supporting other versions, @iCharlesHu have you looked into that yet? If not, is that something we can consider as a followup? We're really trying to land the minimum possible to get this into the toolchain as soon as possible, and since the toolchain only uses 3.24 it'd be great if we could land this first and followup in the near future with expanding support to other versions for testing outside of the toolchain

Reply:
Yeah, that sounds good

Reply:
I think we should be able to use 3.24 for Linux for sure since we build cmake locally right? I guess it's macOS that's the problem...

Reply:
At least the macOS nightly is checking out CMake 3.24.2 as well:

```
 > git init /Users/ec2-user/jenkins/workspace/oss-swift-5.10-package-macos/cmake # timeout=10
Fetching upstream changes from https://github.com/KitWare/CMake.git
 > git --version # timeout=10
 > git --version # 'git version 2.39.3 (Apple Git-145)'
using GIT_ASKPASS to set credentials Github token
Setting http proxy: proxy.config.pcp.local:3128
 > git fetch --tags --force --progress -- https://github.com/KitWare/CMake.git +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
Checking out Revision 31f835410efeea50acd43512eb9e5646a26ea177 (v3.24.2)
```

Are you seeing CI check out something older somewhere?

Reply:
I remember last time we ran into some CI failure.. but can't remember the details. I just bumped it to 3.24 and we'll see if CI fails.